src/vit-based/vit.jl

```diff
@@ -0,0 +1,116 @@
+# Utility function for applying LayerNorm before a block
+prenorm(planes, fn) = Chain(fn, LayerNorm(planes))
+
+"""
+    Transformer(planes, depth, heads, headplanes, mlppanes, dropout = 0.)
+
+Transformer as used in the base ViT architecture.
+([reference](https://arxiv.org/abs/2010.11929)).
+
+# Arguments
+- `planes`: number of input channels
+- `depth`: number of layers
+- `heads`: number of attention heads
+- `headplanes`: number of hidden channels per head
+- `mlppanes`: number of hidden channels in the MLP block
+- `dropout`: dropout rate
+"""
+function Transformer(planes, depth, heads, headplanes, mlpplanes, dropout = 0.)
```

Comment:
This should be `transformer` since it isn't a constructor.

Reply:
And probably with some types & kw:
```suggestion
function transformer(planes::Integer, depth::Integer, heads::Integer, headplanes::Integer, mlpplanes::Integer; dropout::Real = 0.0)
```

Reply:
Is there a more specific name for this than transformer? e.g. the ViT paper calls this part a "transformer encoder".

Reply:
I think `transformer_encoder` is good considering only the rare user will call this directly